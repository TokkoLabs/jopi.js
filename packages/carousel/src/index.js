/* eslint-disable multiline-ternary */
import React, { useEffect, useState } from 'react'
import { Box } from '@oneloop/box'
import theme from '@oneloop/theme'
import { Icon } from '@oneloop/icons'
import '../styles/gallery.css'
import { Text } from '@oneloop/text'

const ButtonGallery = ({ text, ...props }) => (
  <Box
    __css={{
      padding: '6px 12px',
      color: '#FFF',
      borderRadius: '32px',
      background: theme.colors.secondary,
      border: '1px solid #A2D4E5',
      fontFamily: 'Nunito Sans',
      fontSize: '12px',
      fontWeight: '600',
      cursor: 'pointer',
      ':hover': {
        background: '#A2D4E5',
        color: theme.colors.secondary,
      },
      ':active': {
        color: theme.colors.secondaryHover,
        borderColor: theme.colors.secondary,
      },
    }}
    {...props}
  >
    {text}
  </Box>
)

const ImageCard = ({ url, styles, children, ...props }) => {
  const [rectangle, setRectangle] = useState(false)

  const isRectangle = async () => {
    const isRectangle = await validateImageDimensions(url)
    return setRectangle(isRectangle)
  }

  useEffect(() => {
    isRectangle()
  }, [url])

  return (
    <Box
      __css={{
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        background: theme.colors.neutralGray8,
        width: '100%',
        height: '47%',
        borderRadius: '12px',
        backgroundSize: rectangle ? 'cover' : 'contain',
        backgroundPosition: 'center',
        backgroundRepeat: 'no-repeat',
        backgroundImage: `url(${url})`,
        cursor: 'pointer',
        ...styles,
      }}
      {...props}
    >
      {children}
    </Box>
  )
}

const validateImageDimensions = async (imageUrl) => {
  // eslint-disable-next-line no-undef
  const img = new Image()
  img.src = imageUrl

  await new Promise((resolve) => {
    img.onload = () => resolve()
  })

  return img.width > img.height
}

export const Carousel = ({
  images = [],
  planos = [],
  video = [],
  video360 = [],
  otherComponent = false,
  frontCoverImg = false,
  frontCoverBlueprints = false,
  ...props
}) => {
  const [fullscreen, setFullscreen] = useState(false)
  const [tabSelected, setTabSelected] = useState('fotos')
  const tabContainers = []
  const [index, setIndex] = useState(0)
  const [contTab, setContTab] = useState(0)
  const Images = [...images, ...planos]
  const imgWithCover = [...images]
  const bluePrintsWithCover = [...planos]

  if (frontCoverImg) {
    Images.unshift(frontCoverImg)
    imgWithCover.unshift(frontCoverImg)
  }

  if (frontCoverBlueprints) {
    Images.unshift(frontCoverBlueprints)
    bluePrintsWithCover.unshift(frontCoverBlueprints)
    console.log(bluePrintsWithCover)
  }

  const changeTabContainer = (tab) => {
    setIndex(0)
    setTabSelected(tab)
  }

  if (video.length > 0) {
    tabContainers.push('Videos')
  }

  if (video360.length > 0) {
    tabContainers.push('Video360')
  }

  if (images.length > 0 || frontCoverImg.length > 0) {
    tabContainers.push('Fotos')
  }

  if (planos.length > 0 || frontCoverBlueprints.length > 0) {
    tabContainers.push('Planos')
  }

  useEffect(() => {
    let newContTab
    if (tabSelected === 'videos') {
      newContTab = video.length
    } else if (tabSelected === 'fotos') {
      newContTab = imgWithCover.length
    } else if (tabSelected === 'planos') {
      newContTab = bluePrintsWithCover.length
    } else if (tabSelected === 'video360') {
      newContTab = video360.length
    }

    setContTab(newContTab)
  }, [tabSelected])

  const next = () => setIndex((prev) => Math.min(contTab - 1, prev + 1))

  const prev = () => setIndex((prev) => Math.max(0, prev - 1))

  const handleKeyDown = (event) => {
    if (event.key === 'ArrowRight') {
      next()
    }

    if (event.key === 'ArrowLeft') {
      prev()
    }

    if (event.key === 'Escape') {
      closeFullscreen()
    }
  }

  useEffect(() => {
    document.addEventListener('keydown', handleKeyDown)

    return () => {
      document.removeEventListener('keydown', handleKeyDown)
    }
  }, [contTab, index])

  const toggleFullscreen = () => {
    if (
      video.length > 0 ||
      video360.length > 0 ||
      images.length > 0 ||
      planos.length > 0
    ) {
      if (planos.length > 0 && !images.length > 0) {
        setTabSelected('planos')
      }
      setFullscreen(true)
    }
  }

  const closeFullscreen = () => {
    setFullscreen(false)
    setTabSelected('fotos')
    setIndex(0)
  }

  return (
    <Box __css={{ position: 'relative' }}>
      {!otherComponent ? (
        <Box
          {...props}
          __css={{
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'space-between',
            height: '263px',
          }}
        >
          <ImageCard
            onClick={toggleFullscreen}
            className="firstTabImg"
            styles={{
              position: 'relative',
              width: '69%',
              height: '100%',
            }}
            url={Images[0]}
          >
            {!Images[0] && (
              <Icon
                icon="icon-propiedades"
                fontSize="34px"
                color={theme.colors.neutralGray4}
              />
            )}

            <Box
              __css={{
                position: 'absolute',
                bottom: '16px',
                right: '16px',
                display: 'flex',
                gap: '8px',
              }}
            >
              {video.length > 0 && (
                <ButtonGallery
                  text={'Videos'}
                  onClick={() => setTabSelected('videos')}
                />
              )}
              {video360.length > 0 && (
                <ButtonGallery
                  text={'Video 360Â°'}
                  onClick={() => setTabSelected('video360')}
                />
              )}
              {(images.length > 0 || imgWithCover.length > 0) && (
                <ButtonGallery
                  className="buttonGallery"
                  text={'Fotos'}
                  onClick={() => setTabSelected('fotos')}
                />
              )}
              {(planos.length > 0 || frontCoverBlueprints.length > 0) && (
                <ButtonGallery
                  text={'Planos'}
                  onClick={() => setTabSelected('planos')}
                />
              )}
            </Box>
          </ImageCard>

          <Box
            onClick={toggleFullscreen}
            __css={{
              display: 'flex',
              flexDirection: 'column',
              justifyContent: 'space-between',
              width: '29%',
              height: '100%',
            }}
          >
            <ImageCard url={Images[1]}>
              {!Images[1] && (
                <Icon
                  icon="icon-propiedades"
                  fontSize="24px"
                  color={theme.colors.neutralGray4}
                />
              )}
            </ImageCard>

            <ImageCard url={Images[2]}>
              {!Images[2] && (
                <Icon
                  icon="icon-propiedades"
                  fontSize="24px"
                  color={theme.colors.neutralGray4}
                />
              )}
              {images.length > 3 && (
                <Box
                  __css={{
                    width: '48px',
                    height: '48px',
                    display: 'grid',
                    placeItems: 'center',
                    borderRadius: '50%',
                    backgroundColor: theme.colors.neutralGray8,
                    opacity: '0.8',
                    color: theme.colors.black,
                    fontFamily: 'Nunito Sans',
                    fontSize: '14px',
                    fontWeight: '700',
                  }}
                >{`+${Images.length - 3}`}</Box>
              )}
            </ImageCard>
          </Box>
        </Box>
      ) : (
        <Box onClick={toggleFullscreen}>{otherComponent}</Box>
      )}

      <Box className={`fullscreen ${fullscreen ? 'openFullscreen' : ''}`}>
        <Box className="fsOverlay" onClick={closeFullscreen}></Box>
        {tabContainers.length > 1 && (
          <Box className="fsTabHeader">
            {tabContainers.map((tab, i) => (
              <Text
                key={i}
                onClick={() => changeTabContainer(tab.toLocaleLowerCase())}
                className={`${
                  tabSelected === tab.toLocaleLowerCase() && 'tabSelected'
                }`}
                variant="bodyBold.fontSize14"
              >
                {tab}
              </Text>
            ))}
          </Box>
        )}
        <Box className="fsContainerTabs" __css={{ position: 'relative' }}>
          {/* Containers */}

          <Icon
            icon="icon-atras"
            className="iconPrev"
            fontSize="40px"
            onClick={prev}
          />

          {tabSelected === 'fotos' && (
            <Box className="fsTabCont">
              <Box className="fsTabContImage">
                <img src={imgWithCover[index]} alt="Foto" />
                <Text className="contFotos" variant="bodyBold.fontSize14">{`${
                  index + 1
                }/${imgWithCover.length}`}</Text>
              </Box>
            </Box>
          )}

          {tabSelected === 'videos' && (
            <Box className="fsTabCont">
              <Box className="fsTabContImage">
                <iframe
                  width="100%"
                  height="100%"
                  src={video[index]}
                  title="Video"
                  frameBorder="0"
                  allowFullScreen
                />
                <Text variant="bodyBold.fontSize14">{`${index + 1}/${
                  video.length
                }`}</Text>
              </Box>
            </Box>
          )}

          {tabSelected === 'planos' && (
            <Box className="fsTabCont">
              <Box className="fsTabContImage">
                <img src={bluePrintsWithCover[index]} alt="Foto" />
                <Text variant="bodyBold.fontSize14">{`${index + 1}/${
                  bluePrintsWithCover.length
                }`}</Text>
              </Box>
            </Box>
          )}

          {tabSelected === 'video360' && (
            <Box className="fsTabCont">
              <Box className="fsTabContImage">
                <iframe
                  width="100%"
                  height="100%"
                  src={video360[index]}
                  title="Video"
                  frameBorder="0"
                  allowFullScreen
                />
                <Text variant="bodyBold.fontSize14">{`${index + 1}/${
                  video360.length
                }`}</Text>
              </Box>
            </Box>
          )}

          <Icon
            className="iconNext"
            style={{ transform: 'rotate(180deg)' }}
            onClick={next}
            icon="icon-atras"
            fontSize="40px"
          />

          <Box className="fsCloseIcon">
            <Icon
              onClick={closeFullscreen}
              className="closeIcon"
              icon="icon-cerrar"
              fontSize="32px"
            />
          </Box>
        </Box>
      </Box>
    </Box>
  )
}
