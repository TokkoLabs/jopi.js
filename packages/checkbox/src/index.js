import React from 'react'
import { Box } from '@oneloop/box'
import { Svg } from '@oneloop/icons'

const CheckboxBox = ({
  borderColor,
  backgroundColor,
  tick,
  tickColor,
  ...props
}) => (
  <>
    <Svg width={16} height={16} viewBox="0 0 16 16" fill="none" {...props}>
      <rect
        x="0.5"
        y="0.5"
        width="15"
        height="15"
        rx="1.5"
        fill={backgroundColor}
        stroke={borderColor}
      />
      {tick && <path d="M4.5 10L8 13L14 4" stroke={tickColor} />}
    </Svg>
  </>
)

const CheckboxChecked = ({ isAlt, ...props }) => (
  <CheckboxBox
    borderColor={isAlt ? '#D64A36' : '#6A88F2'}
    backgroundColor={isAlt ? '#D64A36' : '#6A88F2'}
    tick
    tickColor={isAlt ? 'white' : 'white'}
    {...props}
  />
)

const CheckboxUnchecked = (props) => (
  <CheckboxBox borderColor="#C1CCD0" backgroundColor="white" {...props} />
)

const CheckboxDisabled = ({ checked = false, ...props }) => (
  <CheckboxBox
    borderColor="#EFF2F3"
    backgroundColor="#EFF2F3"
    tick={checked}
    tickColor="#C1CCD0"
    {...props}
  />
)

const CheckboxIcon = ({ isAlt = false, ...props }) => (
  <React.Fragment>
    <CheckboxChecked
      isAlt={isAlt}
      {...props}
      css={{
        display: 'none',
        'input:checked:not(:disabled) ~ &': {
          display: 'block',
        },
      }}
    />
    <CheckboxUnchecked
      {...props}
      css={{
        display: 'none',
        'input:not(:checked):not(:disabled) ~ &': {
          display: 'block',
        },
      }}
    />
    <CheckboxDisabled
      {...props}
      css={{
        display: 'none',
        'input:disabled:not(:checked) ~ &': {
          display: 'block',
        },
      }}
    />
    <CheckboxDisabled
      checked
      {...props}
      css={{
        display: 'none',
        'input:checked:disabled ~ &': {
          display: 'block',
        },
      }}
    />
  </React.Fragment>
)

export const Checkbox = ({
  isAlt = false,
  sx,
  variant = 'checkbox',
  ...props
}) => (
  <Box sx={{ position: 'relative', width: 18, height: 18 }}>
    <Box
      as="input"
      type="checkbox"
      {...props}
      sx={{
        position: 'absolute',
        opacity: 0,
        overflow: 'hidden',
        width: '100%',
        height: '100%',
      }}
    />
    <Box
      as={CheckboxIcon}
      aria-hidden="true"
      tx="forms"
      variant={variant}
      isAlt={isAlt}
      sx={sx}
      __css={{
        mr: 2,
        borderRadius: 2,
        color: 'gray',
        'input:checked ~ &': {
          color: 'primary',
        },
        'input:focus ~ &': {
          boxShadow: 'active',
        },
      }}
    />
  </Box>
)
