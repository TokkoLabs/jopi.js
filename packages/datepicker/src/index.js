import React, { forwardRef, useRef } from 'react'
import DatePicker from 'react-datepicker'
import '../styles/datepicker.css'
import '@oneloop/theme/styles/globals.css'
import { Box } from '@oneloop/box'
import { Icon } from '@oneloop/icons'
import { Input } from '@oneloop/input'
import { enUS as defaultLocale } from 'date-fns/locale';

const CustomInput = forwardRef(({ value, onClick }, ref) => {
  const inputRef = useRef(ref)
  const handleClick = () => inputRef.current.click()

  return (
    <Input
      style={{ paddingLeft: '34px' }}
      ref={inputRef}
      onClick={onClick}
      prefix={
        <Icon
          icon={'icon-calendario'}
          fontSize="16px"
          onClick={handleClick}
          style={{ cursor: 'pointer' }}
        />
      }
      value={value}
    />
  )
})
CustomInput.displayName = 'CustomInput'

const CustomHeader = ({
  date,
  decreaseMonth,
  increaseMonth,
  prevMonthButtonDisabled,
  nextMonthButtonDisabled,
  locale,
}) => {
  const formatDate = (date, locale) => {
    const month = date.toLocaleString(locale.code, { month: 'long' })
      .replace(/^./, (match) => match.toUpperCase())
    const year = date.toLocaleString(locale.code, { year: 'numeric' })

    return `${month} ${year}`
  }

  return (
    <Box className="customHeader">
      <Box
        as="button"
        className="customHeaderButton"
        onClick={decreaseMonth}
        disabled={prevMonthButtonDisabled}
      >
        <Icon icon='icon-atras' fontSize='16px' color="#2E393F" />
      </Box>
      <Box className="customHeaderDate">
        {formatDate(date, locale)}
      </Box>
      <Box
        as="button"
        className="customHeaderButton"
        onClick={increaseMonth}
        disabled={nextMonthButtonDisabled}
      >
        <Icon icon='icon-siguiente' fontSize='16px' color="#2E393F" />
      </Box>
    </Box>
  )
}

export const Datepicker = (props) => {
  const { locale, ...rest } = props

  const targetLocale = locale ?? defaultLocale

  return (
    <DatePicker
      className="red-border"
      customInput={<CustomInput />}
      renderCustomHeader={headerProps => (
        <CustomHeader {...headerProps} locale={targetLocale} />
      )}
      locale={targetLocale}
      {...rest}
    />
  )
}
