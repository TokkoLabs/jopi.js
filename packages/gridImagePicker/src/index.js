import React from 'react'
import { DndContext, closestCenter, MouseSensor, useSensor, useSensors } from '@dnd-kit/core'
import { SortableContext, rectSortingStrategy } from '@dnd-kit/sortable'
import { Box } from '@oneloop/box'
import ImageItem from './ImageItem'
import useGridImagePicker from './hooks/useGridImagePicker'
import '../styles/GridImagePicker.css'

/**
 * A grid-based image picker component with sortable and selectable images.
 *
 * @param {Object} props - The props for GridImagePicker.
 * @param {string[]} props.listOfSrc - Array of image URLs to display in the grid.
 * @param {number} [props.maxSelectablePreferenceByUser=10] - Maximum number of images that can be selected by the user.
 *        Default is 10. This value is a preference; if there are fewer images available than this number, it will adjust accordingly.
 * @param {number} [props.maxSizeInMB=8] - Maximum size in megabytes for each image. Default is 8 MB. Images exceeding this limit will be marked as an error.
 * @param {number} [props.minAspectRatio] - Minimum aspect ratio allowed for the images. Images that don't meet this ratio will be marked as an error.
 * @param {number} [props.maxAspectRatio] - Maximum aspect ratio allowed for the images. Images that exceed this ratio will be marked as an error.
 * @param {Function} [props.onChange] - Optional callback function invoked whenever the internal state of images changes.
 *        Called with the updated list of items as `(items) => onChange(items)`.
 * @param {Function} [props.sizeFetcher] - Optional fetch function to get the size of the images. Default: (src) => fetch(src)
 * @param {Object} props.texts - Nested texts
 */
export const GridImagePicker = ({
  listOfSrc,
  maxSelectablePreferenceByUser = 10,
  maxSizeInMB = 8,
  minAspectRatio,
  maxAspectRatio,
  onChange,
  texts,
  sizeFetcher,
}) => {
  const { attributes, methods } = useGridImagePicker({
    listOfSrc,
    maxSelectablePreferenceByUser,
    maxSizeInMB,
    minAspectRatio,
    maxAspectRatio,
    onChange,
  })
  const { items, config, status } = attributes
  const {
    handleClickItem,
    handleUpdateItem,
    handleDeselectAll,
    handleSelectAll,
    handleDragStart,
    handleDragEnd,
  } = methods

  const sensors = useSensors(
    useSensor(MouseSensor, {
      activationConstraint: {
        distance: 10,
      },
    })
  )

  return (
    <Box className="gridImagePickerContainer">
      <Box className="gridButtonsWrapper">
        <Box
          as="button"
          className="gridButton gridButtonSelect"
          onClick={handleSelectAll}
          disabled={status.numberOfCheckedItems === config.maxSelectable || !status.itemsAreReady}
        >
          Seleccionar {status.itemsAreReady ? config.maxSelectable : 0}
        </Box>
        <Box
          as="button"
          className="gridButton gridButtonDeselect"
          onClick={handleDeselectAll}
          disabled={status.numberOfCheckedItems === 0}
        >
          Deseleccionar todos
        </Box>
      </Box>

      <DndContext
        sensors={sensors}
        collisionDetection={closestCenter}
        onDragStart={handleDragStart}
        onDragEnd={handleDragEnd}
        autoScroll={false}
      >
        <SortableContext items={items} strategy={rectSortingStrategy}>
          <Box as="section" className="gridImagePicker">
            {items.map((item) => (
              <ImageItem
                key={item.id}
                item={item}
                status={status}
                config={config}
                handleClick={handleClickItem}
                handleUpdateItem={handleUpdateItem}
                texts={texts}
                sizeFetcher={sizeFetcher}
              />
            ))}
          </Box>
        </SortableContext>
      </DndContext>
    </Box>
  )
}

export default GridImagePicker
