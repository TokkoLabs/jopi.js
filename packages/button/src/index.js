import React from 'react'
import { Badge } from '@oneloop/badge'
import { Box } from '@oneloop/box'
import { Checkbox } from '@oneloop/checkbox'
import { Icon } from '@oneloop/icons'
import { Image } from '@oneloop/image'
import theme from '@oneloop/theme'
import '@oneloop/fonts'

export const Button = ({ variant, ...props }) => (
  <Box sx={{ position: 'relative' }}>
    <Box
      as='button'
      tx='buttons'
      type='button'
      variant={variant}
      {...props}
      __css={{
        appearance: 'none',
        display: 'flex',
        lineHeight: 'inherit',
        fontFamily: 'Nunito Sans',
        textAlign: 'center',
        textDecoration: 'none',
        cursor: 'pointer',
        outline: 'none',
        border: 0,
        flexDirection: 'row',
        alignItems: 'center',
        whiteSpace: 'nowrap',
        justifyContent: 'center',
        fontWeight: 'bold',
        borderRadius: '12px',
      }}
    />
  </Box>
)

export const ButtonIcon = ({ icon, variant, text, badgeValue = 0, badgeVariant = ['badgeNormal', 'primary'], maxWidth, srcImage, userImage, hasCheckbox, active, isCollapsible, disabled, holdPress, hover, filled, isExpanded, widthImage, heightImage, setExpand, testForAction, ...props }) => {
  let colorValue
  let colorValueIcon
  let fontSize
  let fontWeight
  let fontSizeIcon
  if (Array.isArray(variant)) {
    const indexes = variant.map(v => Object.keys(theme.buttons).indexOf(v))
    indexes.map(index => {
      const variantValues = Object.values(theme.buttons)[index]
      if (variantValues[':focus'] !== undefined && active) {
        colorValue = variantValues[':focus'].color
        fontWeight = variantValues[':focus'].fontWeight
        if (variantValues.colorIcon !== undefined) {
          colorValueIcon = variantValues[':focus'].colorIcon
        }
      }
      if (colorValue === undefined && variantValues.color !== undefined) {
        colorValue = variantValues.color
      }
      if (colorValueIcon === undefined && variantValues.colorIcon !== undefined) {
        colorValueIcon = variantValues.colorIcon
      }
      if (fontWeight === undefined && variantValues.fontWeight !== undefined) {
        fontWeight = variantValues.fontWeight
      }
      if (fontSize === undefined && variantValues.fontSize !== undefined) {
        fontSize = variantValues.fontSize
      }
      if (filled && colorValueIcon === undefined) {
        colorValueIcon = variantValues.colorFilled
      }
      if (fontSizeIcon === undefined) {
        fontSizeIcon = variantValues.fontSizeIcon
      }

      return colorValue
    })
  } else {
    const index = Object.keys(theme.buttons).indexOf(variant)
    const variantValues = Object.values(theme.buttons)[index]
    if (variantValues !== undefined) {
      colorValue = variantValues.color
      colorValueIcon = variantValues.colorIcon ||
        (
          filled && colorValueIcon === undefined
            ? variantValues.colorFilled
            : variantValues.color
        )
      fontSize = variantValues.fontSize
      fontWeight = variantValues.fontWeight
      if (hover) {
        colorValue = variantValues[':hover'].color
        colorValueIcon = variantValues[':hover'].colorIcon
        fontWeight = variantValues[':hover'].fontWeight
      } else if (active) {
        colorValue = variantValues[':focus'].color
        colorValueIcon = variantValues[':focus'].colorIcon
        fontWeight = variantValues[':focus'].fontWeight
      }
      if (fontSizeIcon === undefined) {
        fontSizeIcon = variantValues.fontSizeIcon
      }
    }
  }
  if (badgeValue > 99) {
    badgeValue = '+99'
  }

  if (testForAction) {
    console.log('It has a boolean testForAction.')
  }

  return (
    <Box sx={ maxWidth ? { position: 'relative', width: '100%' } : { position: 'relative' }}>
      <Box
        as='button'
        tx='buttons'
        type='button'
        variant={variant}
        {...props}
        __css={{
          appearance: 'none',
          display: 'flex',
          lineHeight: 'inherit',
          fontFamily: 'Nunito Sans',
          fontWeight: fontWeight || 'bold',
          textAlign: 'center',
          textDecoration: 'none',
          cursor: 'pointer',
          outline: 'none',
          border: 0,
          flexDirection: 'row',
          justifyContent: !maxWidth && 'center',
          width: maxWidth && '100%',
          alignItems: 'center',
          whiteSpace: 'nowrap',
          position: 'relative',
        }}
      >
        { hasCheckbox && active && <Checkbox defaultChecked disabled={ disabled }/>}
        { hasCheckbox && !active && <Checkbox defaultChecked={false} disabled={ disabled }/>}
        { icon && <Icon icon={icon} fontSize={fontSizeIcon || fontSize} style={ holdPress && { color: (colorValueIcon || colorValue) }}/> }
        { !userImage && srcImage && <Image src={srcImage} style={{ width: widthImage, height: heightImage }}/> }
        { userImage && srcImage && <Image src={srcImage} variant="avatar"/> }
        { userImage && !srcImage && <span className='icon-contactos' style={{ fontSize: '24px' }}/>}
        { text && !holdPress && <span>{text}</span>}
        { text && holdPress && <span style={colorValue && { color: colorValue, fontWeight: fontWeight }}>{text}</span> }
        { isCollapsible && setExpand === undefined && <span className='icon-dropdown' style={{ position: 'absolute', right: '10px', color: '#798B97', fontSize: '14px', transform: isExpanded ? 'rotate(-180deg)' : 'rotate(0deg)', transition: 'all 0.5s', paddingTop: '4px' }}/> }
        { isCollapsible && setExpand !== undefined && <span className='icon-dropdown' onClick={(event) => { event.preventDefault(); setExpand(); event.stopPropagation(); event.nativeEvent.stopImmediatePropagation() }}style={{ position: 'absolute', right: '10px', color: '#798B97', fontSize: '14px', transform: isExpanded ? 'rotate(-180deg)' : 'rotate(0deg)', transition: 'all 0.5s', paddingTop: '4px', paddingLeft: '1px', paddingRight: '1px' }}/> }
        { badgeValue !== 0 && !text && <Badge variant={badgeVariant} isNotButton style={{ position: 'absolute', top: '2px', left: '16px' }}>{badgeValue}</Badge> }
        { badgeValue !== 0 && text && <Badge variant={badgeVariant} isNotButton style={{ position: 'absolute', right: isCollapsible ? '30px' : '10px' }}>{badgeValue}</Badge> }
      </Box>
    </Box>
  )
}
