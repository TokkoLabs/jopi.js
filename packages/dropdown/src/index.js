import React, { createContext, useContext } from 'react'
import { Box } from '@oneloop/box'
import { Button, ButtonIcon } from '@oneloop/button'
import { Icon } from '@oneloop/icons'
import { List } from '@oneloop/list'
import { useToggle, useOnClickOutside } from '@oneloop/hooks'
import { Text } from '@oneloop/text'
import theme from '@oneloop/theme'

const DropdownContext = createContext()

export const Dropdown = ({ children, ...props }) => {
  const ref = React.useRef()
  const [open, toggle] = useToggle(false)

  useOnClickOutside(ref, () => toggle(false))

  const value = React.useMemo(() => ({ open, toggle, ref }), [open])

  return (
    <DropdownContext.Provider value={value}>
      <Box {...props} __css={{ position: 'relative', height: '200px' }}>
        {children}
      </Box>
    </DropdownContext.Provider>
  )
}

const useDropdownContext = () => {
  const context = useContext(DropdownContext)
  if (!context) {
    throw new Error(
      'Dropdown compound components cannot be rendered outside the Dropdown component'
    )
  }
  return context
}

const DropdownButton = ({ icon, text, variant = 'dropdown', disabled = false, filled = false, isButtonIcon = false, isArrowStatic = false, variantSize = 'dropdownSizeNormal', selection, ...props }) => {
  const { toggle } = useDropdownContext()

  const variantValues = Object.values(theme.buttons)[Object.keys(theme.buttons).indexOf(variant)]
  const colorFilled = variantValues.colorFilled
  const backgroundColorFilled = variantValues.backgroundColorFilled
  const colorArrowFilled = variantValues.colorArrowFilled
  const borderFilled = variantValues.borderFilled
  let sizeArrow = variantValues.sizeArrow
  if (sizeArrow === undefined) {
    sizeArrow = '12px'
  }
  // ESTO FALLA CON LOS CASOS QUE MENCIONO MARIAN
  // Porque llamo al buttonIcon
  if (isButtonIcon) {
    return (
      <ButtonIcon
        variant={[variant, variantSize]}
        icon={icon}
        filled={filled}
        holdPress
        onClick={ !disabled ? toggle : undefined}
      />
    )
  }

  return (
    <Button
      variant={[variant, variantSize]}
      {...props}
      onClick={ !disabled ? toggle : undefined}
      sx={{
        display: 'flex',
        flexDirection: 'row',
        alignItems: 'center',
        textOverflow: 'ellipsis',
        fontFamily: 'Nunito Sans',
        overflow: 'hidden',
        whiteSpace: 'nowrap',
        textAlign: 'start',
        color: filled && colorFilled,
        border: filled && borderFilled,
        backgroundColor: filled && backgroundColorFilled,
        justifyContent: isArrowStatic ? 'center' : 'flex-start',
      }}
    >
      { icon && (!filled ? <Icon icon={icon} fontSize='16px'/> : <Icon icon={icon} fontSize='16px' style={{ color: colorFilled }}/>) }
      { !filled ? <span>{text}</span> : <span style={{ color: colorFilled }}>{text}</span> }
      { selection && <span style={ disabled ? { color: '#AEBAC0', fontWeight: 700 } : { color: '#384248', fontWeight: 700 }}>{selection}</span> }
      { !filled
        ? <Icon icon='icon-dropdown' fontSize={sizeArrow} style={{ position: isArrowStatic ? 'static' : 'absolute', right: '12px', transform: 'rotate(0deg)' }}/>
        : <Icon icon='icon-dropdown' fontSize={sizeArrow} style={{ position: isArrowStatic ? 'static' : 'absolute', right: '12px', transform: 'rotate(0deg)', color: colorArrowFilled }}/> }
    </Button>
  )
}

const DropdownList = ({ children, width = '236px', height = '150px', ...props }) => {
  const { open, ref, toggle } = useDropdownContext()
  return (
    open && (
      <List
        onClick={toggle}
        ref={ref}
        {...props}
        sx={{
          display: 'flex',
          flexDirection: 'column',
          flexGrow: 1,
          py: '5px',
          backgroundColor: 'white',
          boxShadow: 'active',
          borderRadius: 'input',
          position: 'absolute',
          top: '38px',
          zIndex: 1,
          width: width,
          height: height,
        }}
      >
        {children}
      </List>
    )
  )
}

const DropdownListSearch = (props) => <List.Search {...props} />

const DropdownListItem = (props) => (
  <List.Item
    {...props}
    sx={{
      cursor: 'pointer',
      fontSize: '2',
      color: 'darkGray',
    }}
  />
)

const DropdownListDefault = ({ disabled, variantSize = 'dropdownSizeNormal', ...props }) => (
  <List.Default
    disabled={disabled}
    variantSize={variantSize}
    {...props}
    sx={{
      cursor: disabled ? 'default' : 'pointer',
      pointerEvents: disabled ? 'none' : undefined,
      userSelect: disabled ? 'none' : undefined,
      fontSize: '2',
      color: 'darkGray',
    }}
  />
)

const DropdownListMultiselect = ({ disabled, active, variantSize = 'dropdownSizeNormal', ...props }) => (
  <List.Multiselect
    disabled={disabled}
    isActive={active}
    variantSize={variantSize}
    {...props}
    sx={{
      cursor: disabled ? 'default' : 'pointer',
      pointerEvents: disabled ? 'none' : undefined,
      userSelect: disabled ? 'none' : undefined,
      fontSize: '2',
      color: 'darkGray',
    }}
  />
)

const DropdownListIcon = ({ disabled, active, icon, variantSize = 'dropdownSizeNormal', ...props }) => (
  <List.Icon
    disabled={disabled}
    icon={icon}
    variantSize={variantSize}
    {...props}
    sx={{
      cursor: disabled ? 'default' : 'pointer',
      pointerEvents: disabled ? 'none' : undefined,
      userSelect: disabled ? 'none' : undefined,
      fontSize: '2',
      color: 'darkGray',
    }}
  />
)

const DropdownText = ({ children }) => (
  <Text variant='body.fontSize12' style={{ padding: '9px 12px 9px 12px' }}>
    {children}
  </Text>
)

Dropdown.Button = DropdownButton
Dropdown.Items = DropdownList
Dropdown.Item = DropdownListItem
Dropdown.Default = DropdownListDefault
Dropdown.Multiselect = DropdownListMultiselect
Dropdown.Icon = DropdownListIcon
Dropdown.Text = DropdownText
Dropdown.Search = DropdownListSearch
