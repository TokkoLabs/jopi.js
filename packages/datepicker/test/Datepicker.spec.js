import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import { Datepicker } from '../src'

describe('Datepicker', () => {
  test('default', () => {
    const tree = renderer.create(<Datepicker></Datepicker>).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
