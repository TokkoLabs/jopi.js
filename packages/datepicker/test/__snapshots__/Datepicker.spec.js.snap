// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Datepicker default 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: relative;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 6px;
  color: #344149;
  font-family: Nunito Sans;
  font-weight: 400;
}

.c1 > *:first-child {
  position: absolute;
  left: 12px;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: grid;
  place-items: center;
  width: 16px;
  height: 16px;
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-weight: 400;
  border: solid 1px transparent;
  box-shadow: none;
  font-family: primary;
  display: block;
  padding-left: 30px;
  padding-right: 12px;
  outline: none;
  width: 100%;
}

@media screen and (min-width:40em) {

}

<div
  className="react-datepicker-wrapper"
>
  <div
    className="react-datepicker__input-container"
  >
    <div
      className="c0"
      width="100%"
    >
      <div
        className="c1"
      >
        <div
          className="c2"
          onClick={[Function]}
          style={
            {
              "cursor": "pointer",
            }
          }
        >
          <i
            className="icon-calendario"
            style={
              {
                "fontSize": "16px",
                "height": "16px",
                "width": "16px",
              }
            }
          />
        </div>
        <input
          className="c3"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          style={
            {
              "paddingLeft": "34px",
            }
          }
          type="text"
          value=""
          width="100%"
        />
      </div>
    </div>
  </div>
</div>
`;
